`cargo bench` will produce a compile error of titan-sys

**Describe this problem**
Run the following command under repo root dir will produce a compile error.
```shell
cargo bench -p benchmarks
```

**Steps to reproduce**
See above.

**Expected behavior**
can run benchmark

**Additional Information**

system: Linux ruihang 5.17.7-arch1-1
cpu: x86_64 Intel(R) Core(TM) i5-9500 CPU @ 3.00GHz

<details>
<summary> Compile Error Log </summary>

```
error: failed to run custom build command for `libtitan_sys v0.0.1 (https://github.com/tikv/rust-rocksdb.git?branch=tikv-5.2#23bd00d5)`

Caused by:
  process didn't exit successfully: `/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-acd593915e6f3c6c/build-script-build` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = Some("/home/ruihang/repo/CeresDB/target/release/build/libz-sys-7060968b1392aa90/out/build")
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: "cmake" "-Wdev" "--debug-output" "/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan" "-DPORTABLE=ON" "-DROCKSDB_DIR=/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb" "-DWITH_TITAN_TESTS=OFF" "-DWITH_TITAN_TOOLS=OFF" "-DWITH_ZLIB=ON" "-DWITH_BZ2=ON" "-DWITH_LZ4=ON" "-DWITH_ZSTD=ON" "-DWITH_SNAPPY=ON" "-DWITH_TITAN_TESTS=OFF" "-DWITH_TITAN_TOOLS=OFF" "-DCMAKE_INSTALL_PREFIX=/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON"
  Running with debug output on.
  -- Enabling RTTI in Debug builds only (default)
     Called from: [2]   /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/cmake/rocksdb_flags.cmake
                  [1]   /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/CMakeLists.txt
  -- Configuring done
  -- Generating /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build
     Called from: [1]   /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/CMakeLists.txt
  -- Generating done
  -- Build files have been written to: /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build
  running: "cmake" "--build" "." "--target" "titan" "--config" "RelWithDebInfo" "--parallel" "6"
  /usr/bin/cmake -S/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan -B/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build --check-build-system CMakeFiles/Makefile.cmake 0
  /usr/bin/make  -f CMakeFiles/Makefile2 titan
  make[1]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  /usr/bin/cmake -S/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan -B/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build --check-build-system CMakeFiles/Makefile.cmake 0
  /usr/bin/cmake -E cmake_progress_start /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles 25
  /usr/bin/make  -f CMakeFiles/Makefile2 CMakeFiles/titan.dir/all
  make[2]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  /usr/bin/make  -f CMakeFiles/titan_build_version.dir/build.make CMakeFiles/titan_build_version.dir/depend
  make[3]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  cd /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan_build_version.dir/DependInfo.cmake --color=
  Dependencies file "CMakeFiles/titan_build_version.dir/titan_build_version.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan_build_version.dir/compiler_depend.internal".
  Consolidate compiler generated dependencies of target titan_build_version
  make[3]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  /usr/bin/make  -f CMakeFiles/titan_build_version.dir/build.make CMakeFiles/titan_build_version.dir/build
  make[3]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  [  4%] Building CXX object CMakeFiles/titan_build_version.dir/titan_build_version.cc.o
  /usr/bin/c++ -DBZIP2 -DHAVE_PCLMUL -DHAVE_SSE42 -DLZ4 -DOS_LINUX -DROCKSDB_FALLOCATE_PRESENT -DROCKSDB_LIB_IO_POSIX -DROCKSDB_MALLOC_USABLE_SIZE -DROCKSDB_PLATFORM_POSIX -DROCKSDB_PTHREAD_ADAPTIVE_MUTEX -DROCKSDB_RANGESYNC_PRESENT -DROCKSDB_SCHED_GETCPU_PRESENT -DROCKSDB_SUPPORT_THREAD_LOCAL -DSNAPPY -DZLIB -DZSTD -I/home/ruihang/repo/CeresDB/target/release/build/snappy-sys-26a487dcbddf5af5/out/build -I/home/ruihang/repo/CeresDB/target/release/build/bzip2-sys-dee24ea472afde1a/out/include -I/home/ruihang/repo/CeresDB/target/release/build/lz4-sys-96e216fd13bc9b20/out/include -I/home/ruihang/repo/CeresDB/target/release/build/libz-sys-7060968b1392aa90/out/include -I/home/ruihang/repo/CeresDB/target/release/build/zstd-sys-b02e0fb93bcc16f8/out/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/util -ffunction-sections -fdata-sections -fPIC -m64 -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -Wno-unused-variable -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers -Wno-strict-aliasing -std=c++11 -fno-omit-frame-pointer -momit-leaf-frame-pointer -Werror -fno-builtin-memcmp -O2 -g -DNDEBUG -MD -MT CMakeFiles/titan_build_version.dir/titan_build_version.cc.o -MF CMakeFiles/titan_build_version.dir/titan_build_version.cc.o.d -o CMakeFiles/titan_build_version.dir/titan_build_version.cc.o -c /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/titan_build_version.cc
  make[3]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  [  4%] Built target titan_build_version
  /usr/bin/make  -f CMakeFiles/titan.dir/build.make CMakeFiles/titan.dir/depend
  make[3]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  cd /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build /home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/DependInfo.cmake --color=
  Dependencies file "CMakeFiles/titan.dir/src/base_db_listener.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_builder.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_cache.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_iterator.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_reader.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_set.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_file_size_collector.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_format.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_gc.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_gc_job.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_gc_picker.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/blob_storage.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/db.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/db_impl.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/db_impl_files.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/db_impl_gc.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/options.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/table_builder.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/table_factory.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/titan_checkpoint_impl.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/titan_stats.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/util.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Dependencies file "CMakeFiles/titan.dir/src/version_edit.cc.o.d" is newer than depends file "/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build/CMakeFiles/titan.dir/compiler_depend.internal".
  Consolidate compiler generated dependencies of target titan
  make[3]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  /usr/bin/make  -f CMakeFiles/titan.dir/build.make CMakeFiles/titan.dir/build
  make[3]: Entering directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  [  8%] Building CXX object CMakeFiles/titan.dir/src/db_impl.cc.o
  /usr/bin/c++ -DBZIP2 -DHAVE_PCLMUL -DHAVE_SSE42 -DLZ4 -DOS_LINUX -DROCKSDB_FALLOCATE_PRESENT -DROCKSDB_LIB_IO_POSIX -DROCKSDB_MALLOC_USABLE_SIZE -DROCKSDB_PLATFORM_POSIX -DROCKSDB_PTHREAD_ADAPTIVE_MUTEX -DROCKSDB_RANGESYNC_PRESENT -DROCKSDB_SCHED_GETCPU_PRESENT -DROCKSDB_SUPPORT_THREAD_LOCAL -DSNAPPY -DZLIB -DZSTD -I/home/ruihang/repo/CeresDB/target/release/build/snappy-sys-26a487dcbddf5af5/out/build -I/home/ruihang/repo/CeresDB/target/release/build/bzip2-sys-dee24ea472afde1a/out/include -I/home/ruihang/repo/CeresDB/target/release/build/lz4-sys-96e216fd13bc9b20/out/include -I/home/ruihang/repo/CeresDB/target/release/build/libz-sys-7060968b1392aa90/out/include -I/home/ruihang/repo/CeresDB/target/release/build/zstd-sys-b02e0fb93bcc16f8/out/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/include -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src -I/home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/util -ffunction-sections -fdata-sections -fPIC -m64 -W -Wextra -Wall -Wsign-compare -Wshadow -Wno-unused-parameter -Wno-unused-variable -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers -Wno-strict-aliasing -std=c++11 -fno-omit-frame-pointer -momit-leaf-frame-pointer -Werror -fno-builtin-memcmp -O2 -g -DNDEBUG -MD -MT CMakeFiles/titan.dir/src/db_impl.cc.o -MF CMakeFiles/titan.dir/src/db_impl.cc.o.d -o CMakeFiles/titan.dir/src/db_impl.cc.o -c /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc
  make[3]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  make[2]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'
  make[1]: Leaving directory '/home/ruihang/repo/CeresDB/target/release/build/libtitan_sys-d9fa18b8ce830101/out/build'

  --- stderr
  make: warning: -j6 forced in submake: resetting jobserver mode.
  In file included from /usr/include/c++/12.1.0/atomic:41,
                   from /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/db/db_impl/db_impl.h:11,
                   from /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.h:3,
                   from /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:1:
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual rocksdb::Status rocksdb::titandb::TitanDBImpl::FileManager::BatchFinishFiles(uint32_t, const std::vector<std::pair<std::shared_ptr<rocksdb::titandb::BlobFileMeta>, std::unique_ptr<rocksdb::titandb::BlobFileHandle> > >&)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:67:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 153 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  In file included from /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:17:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual rocksdb::Status rocksdb::titandb::TitanDBImpl::FileManager::BatchFinishFiles(uint32_t, const std::vector<std::pair<std::shared_ptr<rocksdb::titandb::BlobFileMeta>, std::unique_ptr<rocksdb::titandb::BlobFileHandle> > >&)’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::SeekToLast()’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:64:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 146 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::SeekToLast()’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::Prev()’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:102:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 148 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::Prev()’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::Next()’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:92:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 147 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::Next()’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::SeekToFirst()’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:55:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 146 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::SeekToFirst()’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::Seek(const rocksdb::Slice&)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:73:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 146 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::Seek(const rocksdb::Slice&)’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual void rocksdb::titandb::TitanDBIterator::SeekForPrev(const rocksdb::Slice&)’ at /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/titan/src/db_iter.h:82:17:
  /usr/include/c++/12.1.0/bits/atomic_base.h:618:35: error: array subscript 146 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
        |                                   ^~~~~
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h: In member function ‘virtual void rocksdb::titandb::TitanDBIterator::SeekForPrev(const rocksdb::Slice&)’:
  /home/ruihang/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/23bd00d/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:80:31: note: while referencing ‘rocksdb::StatisticsImpl<145, 49>::StatisticsData::tickers_’
     80 |     std::atomic_uint_fast64_t tickers_[TICKER_MAX];
        |                               ^~~~~~~~
  cc1plus: all warnings being treated as errors
  make[3]: *** [CMakeFiles/titan.dir/build.make:261: CMakeFiles/titan.dir/src/db_impl.cc.o] Error 1
  make[2]: *** [CMakeFiles/Makefile2:88: CMakeFiles/titan.dir/all] Error 2
  make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/titan.dir/rule] Error 2
  make: *** [Makefile:172: titan] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /home/ruihang/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.48/src/lib.rs:975:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
```

</details>

have the same error 
CPU Intel(R) Core(TM) i5-4300U CPU @ 1.90GHz
Fedora Linux 36 (Server Edition)

```
gmake: warning: -j4 forced in submake: resetting jobserver mode.
  In file included from /usr/include/c++/12/atomic:41,
                   from /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/../rocksdb/db/db_impl/db_impl.h:11,
                   from /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/titan/src/db_impl.h:3,
                   from /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:1:
  In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::fetch_add(__int_type, std::memory_order) [with _ITp = long unsigned int]’,
      inlined from ‘void rocksdb::StatisticsImpl<TICKER_MAX, HISTOGRAM_MAX>::recordTick(uint32_t, uint64_t) [with unsigned int TICKER_MAX = 145; unsigned int HISTOGRAM_MAX = 49]’ at /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics_impl.h:205:59,
      inlined from ‘void rocksdb::RecordTick(Statistics*, uint32_t, uint64_t)’ at /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/../rocksdb/monitoring/statistics.h:33:27,
      inlined from ‘virtual rocksdb::Status rocksdb::titandb::TitanDBImpl::FileManager::BatchFinishFiles(uint32_t, const std::vector<std::pair<std::shared_ptr<rocksdb::titandb::BlobFileMeta>, std::unique_ptr<rocksdb::titandb::BlobFileHandle> > >&)’ at /home/paomian/.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/7737841/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:67:17:
  /usr/include/c++/12/bits/atomic_base.h:618:35: error: array subscript 153 is above array bounds of ‘std::atomic_uint_fast64_t [145]’ {aka ‘std::atomic<long unsigned int> [145]’} [-Werror=array-bounds]
    618 |       { return __atomic_fetch_add(&_M_i, __i, int(__m)); }
```

--- a/wal/Cargo.toml
+++ b/wal/Cargo.toml
@@ -26,5 +26,5 @@ env_logger = "0.6"
 
 [dependencies.rocksdb]
 git = "https://github.com/tikv/rust-rocksdb.git"
-rev = "6f263ec5638835eda5911a34046fc303bc307dc7"
+rev = "084102f7e4d1901cbe3f2782c5c63cb7af628bac" # at branch tikv-6.1
 features = ["portable"]

